javascript/shoppinglist.js

```diff
@@ -1,38 +1,175 @@
-const getShoppingList = () => {
-	const ShoppingList = {
-		list: [],
-		addToList: (item) => {
-			ShoppingList.list.push(item);
+"use strict";
+(function () {
+	const inpQtyElm = document.querySelector('.inp-qty .inp');
+	const inpNameElm = document.querySelector('.inp-item-name .inp');
+	const inpPriceElm = document.querySelector('.inp-item-price .inp');
 
-			return ShoppingList.list;	
-		},
-		displayList: () => {
-			let str = '';
-			for (let i = 0; i < ShoppingList.list.length; i++) {
-				str += (i+1) + '. ' + ShoppingList.list[i] + '\n';
+	const inpEditIdElm = document.querySelector('#edit-row-id');
+
+	const btnAdd = document.querySelector('.inp-actions .btn-add');
+	const btnEdit = document.querySelector('.inp-actions .btn-edit');
+	const btnCancel = document.querySelector('.inp-actions .btn-cancel');
+
+	const ListObj = (function () {
+		const _listBody = document.querySelector('.lst-items');
+		const totalQty = document.querySelector('.lst-totals .total-qty');
+		const totalPrice = document.querySelector('.lst-totals .total-price');
+		const listTitleElm = document.querySelector('.list .list-title');
+
+		const LOCAL_STORAGE_LIST = 'list';
+
+		let _items = null;
+		let listName = '';
+		let _mode = 'add';
+
+		const _render = () => {
+			let _qtyTotal = 0;
+			let _priceTotal = 0;
+
+			_listBody.innerHTML = '';
+
+			listTitleElm.innerHTML = listName;
+
+			if (_mode === 'add') {
+				btnAdd.classList.remove('hidden');
+				btnEdit.classList.add('hidden');
+				btnCancel.classList.add('hidden');
+
+			} else if (_mode === 'edit') {
+				btnAdd.classList.add('hidden');
+				btnEdit.classList.remove('hidden');
+				btnCancel.classList.remove('hidden');
+			}
+
+			_items.forEach(item => {
+				_qtyTotal = _qtyTotal + item._qty;
+				_priceTotal = _priceTotal + item._price;
+				_listBody.appendChild(item.render());
+			});
+
+			totalQty.innerHTML = _qtyTotal;
+			totalPrice.innerHTML = _priceTotal.toFixed(2);
+		};
+
+		const _deepFreeze = () => {
+			console.log('deep freeze');
+			if (!window.hasOwnProperty('localStorage')) {
+				console.log('Cannot save to long term storage as this browser does not support it.');
+				return;
+			}
+			window.localStorage.setItem(LOCAL_STORAGE_LIST, JSON.stringify(_items));
+		};
+
+		const _deepDefrost = () => {
+			console.log('deep defrost');
+			if (!window.hasOwnProperty('localStorage')) {
+				console.log('Cannot read from long term storage as this browser does not support it.');
+				return [];
+			}
+
+			const data = JSON.parse(window.localStorage.getItem(LOCAL_STORAGE_LIST));
+			if (!data || data.length < 1) {
+				return [];
+			}
+
+			data.forEach(item => {
+				// TODO: There must be a better way?
```

Comment:
We can use localstorage as a [JSON object](https://coderwall.com/p/ewxn9g/storing-and-retrieving-objects-with-localstorage-html5) instead of having to iterate to do `setPrototypeOf`.